api_server.py

```
@@ -549,4 +555,12 @@ def add_security_headers(response):
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
+    # Safeguard: Do not run the development server in production.
+    if os.environ.get("FLASK_ENV") == "production":
+        print(
+            "Error: Do not run the development server in production. Use a WSGI server (e.g. gunicorn) instead."
+        )
+        sys.exit(1)
+
+    port = int(os.environ.get("PORT", 5000))
+    app.run(host="0.0.0.0", port=port)
```

Ruff (0.14.14)

[error] Possible binding to all interfaces

(S104)